src/code-util.jl

```diff
@@ -266,6 +269,31 @@ function extract_problems(results)
     return problems
 end
 
+# In some error cases the results may be nothing, rather than empty
+function extract_ics(results::Nothing)
+    return []
+end
+
+function extract_ics(results)
+    ics = []
+
+    if !haskey(results, IC_LINE_KEY)
+        return ics
+    end
+
+    # Diagnostic categories have identical ordering so we can use row to find matches
+    # across categories, starting with row 1
+    for i in 1:length(results[IC_LINE_KEY][1])
+        line = extract_detail(results, IC_LINE_KEY, 2, i)
+        report = extract_detail(results, IC_REPORT_KEY, 2, i)
```

Comment:
could you also add the values to this print out if available?

Reply:
if the ic is written as `ic foo(x) { bar(x) implies Int(x) }` then any values of `x` that violate the IC will be reported